src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerAssemblyLoader.Core.cs

```diff
@@ -244,6 +249,31 @@ internal sealed class CompilerAnalyzerAssemblyResolver(AssemblyLoadContext compi
                 }
             }
         }
+
+        private sealed class DiskResolver : IAnalyzerAssemblyResolver
+        {
+            public static readonly DiskResolver Instance = new DiskResolver();
+            public Assembly? Resolve(AnalyzerAssemblyLoader loader, AssemblyName assemblyName, AssemblyLoadContext directoryContext, string directory)
+            {
+                var realPath = loader.GetAssemblyLoadPath(assemblyName, directory);
+                return realPath is not null ? directoryContext.LoadFromAssemblyPath(realPath) : null;
+            }
+        }
+
+        private sealed class StreamResolver : IAnalyzerAssemblyResolver
+        {
+            public static readonly StreamResolver Instance = new StreamResolver();
```

Comment:
```suggestion
            public static readonly StreamResolver Instance = new StreamResolver();
            
```

and same points here as above.